File: fxgl/src/main/java/com/almasb/fxgl/gameplay/AchievementListener.java
/*
 * FXGL - JavaFX Game Library. The MIT License (MIT).
 * Copyright (c) AlmasB (almaslvl@gmail.com).
 * See LICENSE for details.
 */

package com.almasb.fxgl.gameplay;

/**
 * Marks a service that wants to listen for achievement events.
 *
 * @author Almas Baimagambetov (almaslvl@gmail.com)
 */
public interface AchievementListener {

    /**
     * Fired on achievement event.
     *
     * @param event the event
     */
    void onAchievementEvent(AchievementEvent event);
}

File: fxgl-base/src/main/kotlin/com/almasb/fxgl/gameplay/achievement/AchievementListener.kt
/*
 * FXGL - JavaFX Game Library. The MIT License (MIT).
 * Copyright (c) AlmasB (almaslvl@gmail.com).
 * See LICENSE for details.
 */

package com.almasb.fxgl.gameplay.achievement

/**
 * Listener for achievement events.
 *
 * @author Almas Baimagambetov (almaslvl@gmail.com)
 */
interface AchievementListener {

    /**
     * Fired on achievement event.
     *
     * @param event the event
     */
    fun onAchievementEvent(event: AchievementEvent)
}